src/components/Courses/Courses.jsx

```diff
@@ -0,0 +1,166 @@
+ * @returns {string}
+ */
+let getAuthorsNames = (authorIDs, authorsList) => {
+	let names = [];
+	authorIDs.forEach((id) => {
+		let foundAuthor = authorsList.find((e) => e.id === id);
+		names.push(foundAuthor.name);
+	});
+	return names.join(', ');
+};
+
+/**
+ * Search courses by Title or ID
+ *
+ * @returns Object[]
+ */
+let getSearchedCourses = (coursesList) => {
+	// Get Search input value
+	let inputData = document.getElementById('search').value;
+	inputData = inputData.toLowerCase();
+
+	// The array with search result
+	let searchResult = [];
+
+	// Looking for courses by Title or ID
+	coursesList.forEach((course) => {
+		let courseTitle = course.title.toLowerCase();
+		let courseId = course.id.toLowerCase();
+		if (courseTitle.includes(inputData) || courseId.includes(inputData)) {
+			// If found, add to result array
+			searchResult.push(course);
+		}
+	});
+
+	return searchResult;
+};
+
+/**
+ * Courses React component
+ */
+function Courses(props) {
+	const [coursesList, setCoursesList] = useState(
+		props.coursesList ? props.coursesList : mockedCoursesList
+	);
+	const [authorsList, setAuthorsList] = useState(
+		props.authorsList ? props.authorsList : mockedAuthorsList
+	);
+	const [coursesHidden, setCoursesHidden] = useState(props.hidden);
+	const [createCourseFormDisplay, setCreateCourseFormDisplay] = useState(false);
+	const [saveCoursesList, setSaveCoursesList] = useState(coursesList);
+
+	/**
+	 * Find courses by Search input value
+	 */
+	let findCourses = () => {
+		setCoursesList(getSearchedCourses(saveCoursesList));
+	};
+
+	/**
+	 * Display all Courses if Search input is clear
+	 */
+	let clerFilter = () => {
+		// Get Search input value
+		let inputData = document.getElementById('search').value;
+
+		if (!inputData) {
+			setCoursesList(saveCoursesList);
+		}
+	};
+
+	/**
+	 * Render the Create Course Form
+	 */
+	let renderCreateCourseForm = () => {
+		setCoursesHidden(true);
+		setCreateCourseFormDisplay(true);
+	};
+
+	if (!coursesHidden) {
+		return (
+			<div className='courses-wrapper'>
+				<div className='row mt-4'>
+					<div className='col-lg-7 mb-4 mb-lg-0'>
+						<SearchBar actionButton={findCourses} actionInput={clerFilter} />
+					</div>
+					<div className='col-lg-5 text-end'>
+						<Button
+							buttonClass='btn btn-outline-success'
+							buttonText='Add new course'
+							onClick={renderCreateCourseForm}
+						/>
+					</div>
+				</div>
+				<div className='courses'>
+					{coursesList.map((course) => {
+						return (
+							<CourseCard
+								key={course.id}
+								title={course.title}
+								description={course.description}
+								authors={getAuthorsNames(course.authors, authorsList)}
+								duration={course.duration}
+								created={course.creationDate}
+							/>
+						);
+					})}
+				</div>
+			</div>
+		);
+	}
+	if (createCourseFormDisplay) {
+		return (
+			<CreateCourse
+				coursesList={coursesList}
+				authorsList={authorsList}
+				hidden={false}
```

Comment:
Generally you don't need to explicitly pass false here, since if you just omit it it will have value 'undefined' in component which you can check

Reply:
fixed